### ISSUE TITLE
Evac shelter NPC quests are too hard

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#57285]
### Is your feature request related to a problem? Please describe.

So today a new player came into the IRC channel, asking for advice because they did not know how to proceed. The evac shelter NPC sent them into the city center which is impossible for a newly spawned standard survivor, causing them to die multiple times, having to start over.

Other things are just pretty frustrating, imagine finally getting to that house with the dog after two days of preparation, just for the mission to fail as you get near due to the dog getting killed by zeds.

The first impression counts!

### Solution you would like.

This is a tough one. There are basically these solutions:

1. Higher percentage of getting an "easy" quest. For example the cattail one, assuming you get lucky with swamp spawns.
2. Somehow making the NPC you spawn with special, making them give you tasks essential for survival.
2.1 Mission 1: "Make sure there is no imminent danger", checks for zombies in a given radius and wants you to defeat them, but there should be an option of just fleeing to a safer place and trying again.
2.2 Mission 2: "Secure water supply", some quest which requires some tool to boil water and maybe also a little supply of water.
2.3 Mission 3: "Gather food", asks you to forage a bit. Could be mixed with the above mission since usually you have a small amount of water in the evac shelter and forage right away to level up your survival skill.
2.4 Mission 4: "Not just surviving, but thriving", wants you construct idk a proper bed or better weapons, you get the idea. The weapons part could also be part of Mission 1, since cudgels or pointy sticks are essential early game weapons.
3. Tutorial mode. "Are you playing for the first time?", then spawns a tutorial NPC which is described above and optionally lowers the difficulty or something.

I prefer option 2.

### Describe alternatives you have considered.

It is pretty realistic for people to demand the meds they forgot or to worry about their dog. There might be a compromise though: Both missions co-existing.
It would be even better if the mechanism for that can be smart: If the size of that given city is large or there is a great amount of zombies in the vicinity of that, the NPC could say "I am not sure if we could manage to do that, maybe we should prepare first."
On the other hand: Humans are egoistical by default. "I will cooperate with you so we can both survive, but ONLY if you basically commit suicide by going into the city to save my dog" seems logical for some.

**Comment by PatrikLundell** [comment]
Aren't these standard starting NPC quests, i.e. common to all scenarios where you start with an NPC?

People in shock can get lethally fixated on things, so saving a dog that's the only thing left in their life might not seem that unlikely a request. I've never received that one, but I'd expect it to fail most of the time unless you kill off all zombies in the vicinity first, given how stupid "NPC" logic is (like the meteorologist I stumbled upon in my current game: smashing the window and cutting herself while getting out asking what's happening, while completely unarmed and without martial skills).

A tutorial+easily recruitable NPC (as fulfilling quests makes them much easier to recruit) as per 2 should probably be a game option, with the "normal" one spawed otherwise.

**Comment by Zireael07** [comment]
I'd combine 1 and 2 personally (add more easy quests and give them a bigger chance, or even have a priority queue for them so that you need to have done at least 1 easy before hards show up)

**Comment by PatrikLundell** [comment]
Tying quest difficulties into tiers is nearly useless if you upgrade the game frequently and start new worlds rather than continuing in old ones when a character dies. My way of updating is to install a new copy of the game (renaming/removing the previous one), copying the current save only if one exists. That means such data isn't retained. I wouldn't be opposed to a larger percentage of easy quests, though.

**Comment by catdach** [comment]
Something to keep in mind is that the starting NPC isn't meant to be a tutorial for the game (there's already tutorial), so option 3 really shouldn't be considered. That said, I can empathize with the frustration of being given a difficult quest at the very beginning of the game (especially with some quests being literally impossible because of zombies killing the objective). 

Option 1 seems like a bit of a non-solution; Even with the cattail quest; it directs you to the nearest swamp tile which isn't always a safe/accessible place (freaking mosquitos 😡).

So I'm leaning heavily towards something like option 2. 
I'm not sure that a mission like 2.1 is even possible code wise, (what trigger could we even use to say that the mission is done?). Even if it is possible, it may be a bad Idea to have the very first mission ask the player to fight something (brand new players tend to have a bad time fighting things), this also gives the impression that players _should_ be able to take on a few zombies (with their bare hands) right when they spawn. 2.2 and 2.3 seem somewhat redundant as most evac shelters have protein rations and bottled water in the lockers. I get that the idea is to force the player to learn how to be self reliant but I really don't like the idea of the NPC demanding a new player to prove their basic survival skills. we're trying to make a **good** first impression right?

I would like to purpose an option 4
4. Basically option 2 but with slightly different missions

4.1: "hey, can you grab me some food and water from the lockers?" 
This should get new players comfortable with basic looting while also giving them time to get used to other parts of basic gameplay(movement, inventory, needing to walk next to locker to see the stuff inside them, etc.). It also just lets them know that there **is** food and water in the shelter that they can survive off of while they learn to play the game.

4.2: "There should be a med-kit in the shelter somewhere, could you grab me a bandage?"
I'm pretty sure the evac shelters spawn with at least one med kit (in the underground bathrooms IIRC). This also shows that the shelter has some medical supplies (similar to 4.2). 

4.3: "We should make some extra weapons in case we run into something while we're outside the shelter. Could you make a nailboard for me?"
The basic idea of this one is have the player go outside to get a rock, smash one of the wooden benches, and make a nailboard. This mission should probably have a tip like "if you can find a rock outside, you can use that to smash some of these benches and you could even use it hammer some nails into one of the leftover planks."

There could be a few more but that should be enough to start out with. I think the point is to give new players something to do without setting them up for failure, being demanding, or being a literal tutorial. It's great to teach new players to boil water and whatnot but that's actually a pretty complex task for a new player (the evac shelter doesn't even have a place to safely contain a fire lol)

**Comment by PatrikLundell** [comment]
Swamps used to be good places to get some resources. Now you're likely to run into deadly monsters there, as catdach noted.

I believe the med kit drops in the evac shelter are based on probabilities, i.e. each potential location has a probability of dropping a med kit. If so, there is a probability none are present (and I believe I've encountered that case in some start).

**Comment by NetSysFire** [comment]
>Something to keep in mind is that the starting NPC isn't meant to be a tutorial for the game (there's already tutorial)

Yes, but it is far off from actual gameplay. I just tried it (and found some bugs, hehe) and new players would be brutally dropped into the harsh world, experiencing what I already described, which can be quite frustrating.
Maybe another tutorial for just that could be part of the solution. Something which drops you into a walled off area with a little forest, a road to build a simple vehicle and something telling you to forage and go make a makeshift shelter and a cudgel to bash some zed heads in, which is in a locked zone, accessed via a terminal so the zombies can not break out.

>I believe the med kit drops in the evac shelter are based on probabilities,

You are correct.

https://github.com/CleverRaven/Cataclysm-DDA/blob/992dbdcbeb0e583c1bf87d0bd48fe7403a9d3aa9/data/json/itemgroups/SUS/evac_shelter.json#L195

**Comment by leemuar** [comment]
I was that new player too. My "mistake" was to play through quests just like in other RPGs. I struggled and I was killed many times in hospital, bandit camps and other quest locations. New players expect the game to teach them through quests. CDDA's current design is different. It does not try to gradually teach you the world through series of simple quests. It throws you into the world - do whatever you want. 'It's freedom like you never knew", but appreciation of this freedom comes much, much later

The problem I think is new players do not know what they want. They expect quests to give them some ideas and examples, but quests are not designed this way currently.

**Comment by Inglonias** [comment]
> > Something to keep in mind is that the starting NPC isn't meant to be a tutorial for the game (there's already tutorial)
> 
> Yes, but it is far off from actual gameplay. I just tried it (and found some bugs, hehe) and new players would be brutally dropped into the harsh world, experiencing what I already described, which can be quite frustrating. Maybe another tutorial for just that could be part of the solution. Something which drops you into a walled off area with a little forest, a road to build a simple vehicle and something telling you to forage and go make a makeshift shelter and a cudgel to bash some zed heads in, which is in a locked zone, accessed via a terminal so the zombies can not break out.
> 
> > I believe the med kit drops in the evac shelter are based on probabilities,
> 
> You are correct.
> 
> https://github.com/CleverRaven/Cataclysm-DDA/blob/992dbdcbeb0e583c1bf87d0bd48fe7403a9d3aa9/data/json/itemgroups/SUS/evac_shelter.json#L195

It would probably be easier to expand the existing tutorial area, since it already has many of these capabilities.

**Comment by leemuar** [comment]
> starting NPC isn't meant to be a tutorial for the game

There is an argument against that. That start NPC can actually tell you a lot about the game: about faction camps, what have changed in the game, give some nice tip and more. Not a beginner's tutorial, but it was definitely intended to give players a helpful information about the game.
